<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Co-localization Calculations and Statistics • Cygnus</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Co-localization Calculations and Statistics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">Cygnus</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/examples.html">Guided Tutorial</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-full-manual" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Full Manual</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-full-manual">
<li><a class="dropdown-item" href="../articles/data_in_out.html">Data Input and Output</a></li>
    <li><a class="dropdown-item" href="../articles/marker_dist.html">Marker Intensity Distribution Modifications</a></li>
    <li><a class="dropdown-item" href="../articles/marker_co.html">Marker Colocalization and Correlation</a></li>
    <li><a class="dropdown-item" href="../articles/dim_red.html">Dimensionality Reduction</a></li>
    <li><a class="dropdown-item" href="../articles/clustering.html">EV Heterogeneity</a></li>
    <li><a class="dropdown-item" href="../articles/make_report.html">On-demand Report Generation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/yeinchung/Cygnus/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Co-localization Calculations and Statistics</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/yeinchung/Cygnus/blob/master/vignettes/marker_co.Rmd" class="external-link"><code>vignettes/marker_co.Rmd</code></a></small>
      <div class="d-none name"><code>marker_co.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview-of-the-conventional-upset-plot">Overview of the conventional UpSet plot<a class="anchor" aria-label="anchor" href="#overview-of-the-conventional-upset-plot"></a>
</h2>
<p>Upset plot is a visualization method designed to analyze and
interpret set intersections across multiple categories <a href="https://pubmed.ncbi.nlm.nih.gov/26356912/" class="external-link">(Lex, 2014)</a>. Unlike
traditional Venn diagrams, which are limited to fewer than five sets,
UpSet plots are better suited for handling complex datasets with
numerous categories. This method provides a clearer and more
comprehensive representation of intersecting relationships.</p>
<p>This method has found widespread use in various domains. For example,
they are useful in gene set analysis, where researchers examine the
overlap between differentially expressed genes under various
experimental conditions <a href="@article%7Bjensen2017rseqrep,%20author%20=%20%7BJensen,%20T.%20L.%20and%20Frasketi,%20M.%20and%20Conway,%20K.%20and%20others%7D,%20title%20=%20%7BRSEQREP:%20RNA-Seq%20Reports,%20an%20open-source%20cloud-enabled%20framework%20for%20reproducible%20RNA-Seq%20data%20processing,%20analysis,%20and%20result%20reporting%7D,%20journal%20=%20%7BF1000Research%7D,%20year%20=%20%7B2017%7D,%20volume%20=%20%7B6%7D,%20pages%20=%20%7B2162%7D,%20doi%20=%20%7B10.12688/f1000research.13049.1%7D,%20url%20=%20%7Bhttps://doi.org/10.12688/f1000research.13049.1%7D%20%7D" class="external-link">(Jensen,
2017)</a>. UpSet plots are also used in proteomics analysis for
comparing protein identifications in different samples <a href="https://pnnl-comp-mass-spec.github.io/proteomics-data-analysis-tutorial/" class="external-link">(Sagendorf,
2022)</a>. Especially in biological contexts, UpSet plots allow
exploration of complex intersections, providing a useful tool for
analyzing multidimensional data.</p>
<p>At the core of an UpSet plot is a matrix-like structure that
visualizes the intersections. The x-axis typically represents different
combinations of intersecting sets, while the y-axis displays the size of
these intersections (i.e., the cardinalities). Horizontal bars indicate
the total size of each individual set. Dot matrices or line connectors
are used to indicate which sets contribute to a given intersection. This
structured layout allows users to easily discern patterns and
relationships that might be obscured in more traditional
visualizations.</p>
<p>Generally, Interpreting an UpSet plot involves identifying the most
significant intersections, as denoted by the vertical bar plot. The
corresponding dot combinations are then examined to understand the
groups that share many co-occurring elements. For instance, in a study
analyzing movie genres, the largest intersection might involve movies
classified exclusively as Drama, while smaller intersections might
reveal combinations such as Crime and Thriller (see figure <a href="https://upset.app/" class="external-link">here</a>). These insights help highlight the
distribution and overlap between categories. In this example, an
appropriate interpretation would be that most movies belong to a single
genre, while fewer movies are shared across multiple genres.</p>
<p>One of the key strengths of UpSet plots is their scalability and
ability to handle large and complex datasets. They provide a clear and
intuitive visualization even when dealing with numerous sets and
intersections, which would be difficult or impossible to represent using
traditional Venn diagrams. Moreover, they facilitate quick
identification of significant intersections, making them ideal for
exploratory data analysis.</p>
<p>However, UpSet plots also have limitations. As the number of sets and
intersections increases, the plot can become cluttered, making
interpretation more challenging. Additionally, generating and
customizing UpSet plots often requires specialized software or coding
skills, which might pose a barrier for users without technical
expertise. Lastly, there is currently no standardized method for
integrating statistical analyses to identify significantly enriched or
depleted intersections. Addressing these challenges would enhance the
utility of UpSet plots for broader applications.</p>
<p>##Adding statistical analysis to UpSet</p>
<p>###Concept of deviation</p>
<p>When the UpSet plot was introduced, the concept of deviation was also
discussed <span class="citation">@6876017</span>. Deviation describes
the difference between the expected and observed cardinality. The
expected cardinality is calculated under the assumption that the events
in the intersection are independent, by multiplying the proportions of
the intersecting sets. A high deviation suggests that the sets in the
intersection share more-than-expected co-occurring elements, making it a
measure of “surprise.” However, a high deviation may result from random
chance or from a meaningful dependence between sets. This necessitates a
statistical approach to quantify how likely a deviation value could
arise from randomness.</p>
<p>###Methodology for statistical significance</p>
<p>We propose a method to compute and visualize the statistical
significance of deviations. The process begins with a matrix where rows
indicate individual elements, and columns indicate their set
memberships. For example, an element belonging to sets A and B, but not
C, would be described as having a value of 1 in columns A and B and 0 in
C. For each intersection, we simulate the null distribution of
deviations by randomly permuting each column. This destroys any
potential dependence or relationship between sets, ensuring the
independence assumption is valid when calculating expected
cardinalities. By comparing the observed deviation with the simulated
null distribution, we calculate the likelihood of obtaining such a
deviation by chance. This allows us to compute p-values for each
intersection. We then apply a Bonferroni correction to identify
statistically significant deviations.</p>
<div class="section level3">
<h3 id="example-analysis-and-interpretations">Example analysis and interpretations<a class="anchor" aria-label="anchor" href="#example-analysis-and-interpretations"></a>
</h3>
<p>For our analysis, we used imaging data from a mouse lungcancer cell
lines, A549, H2228, and KP1.9. Each element represents an extracellular
vesicle (EV), and set membership is determined by the expression of lung
cancer-associated markers (e.g., EpCAM, EGFR, SDC1, MET, ADAM10). The
results revealed that some intersections (e.g. ADAM10 and MET) showed
high deviations with p-values below the Bonferroni-corrected threshold.
This indicates that the co-occurrence of these markers is likely
attributable to inherent biological dependence rather than stochastic
processes. Such findings highlight potential underlying biology and
suggest that these markers may serve as reliable biomarkers for
identifying EVs of lung cancer, owing to their consistent and
significant co-expression patterns.</p>
<p>By incorporating statistical significance into UpSet plots, we
provide a more rigorous and interpretable framework for analyzing set
intersections. This method highlights biologically meaningful patterns
that might otherwise be obscured.</p>
<pre><code><span><span class="co">## [1] "Calculating expected and observed probabilities..."</span></span>
<span><span class="co">## [1] "Running simulation to calculate the null distribution of deviations..."</span></span></code></pre>
<p><img src="marker_co_files/figure-html/comparison-1.png" width="1200"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yein Chung.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
